### DISCUSSION TITLE
Search input is disappeared after update

### DISCUSSION THREAD

**Discussion by andreygladilin** [squidfunk/mkdocs-material#4850] [Q&A] [answered]
I use mkdocs material+mike+gitlab pages. Unfortunately, after regular update of my documentation on 10th of  Jan 2023 the Search field is disappeared from my site.

**Comment by squidfunk** [comment]
Thanks for reporting. Might be a bug, might be related to customizations, might be a configuration error. Please first read our [upgrade guide](https://squidfunk.github.io/mkdocs-material/upgrade/#upgrading-from-8x-to-9x) if you haven't, and if that doesn't help, follow our [bug reporting guide](https://squidfunk.github.io/mkdocs-material/bug-report/) which explains everything you need to know.

**Comment by do-me** [comment, marked as answer]
Related issue #4851. 
I have the same problem. The [upgrade guide](https://squidfunk.github.io/mkdocs-material/upgrade/#upgrading-from-8x-to-9x) references that something in the search changes, but not what to adapt.

After digging through the changes, I found that something in the header partial changed the block is looking for . Please see [here, line 139](https://github.com/squidfunk/mkdocs-material/pull/4628/files#diff-70a179faf78bd9f0688cfb1feae409bd13411145b6a1fd63069ca5f7412136c1L139).

@squidfunk 
Please add to your upgrade guide (version jump 8 to 9) what exactly to change in the `mkdocs.yml`. I suppose the search plugin must be called differently or something is missing? 
Tried to fix it by comparing it to mkdocs-material `mkdocs.yml` itself, but couldn't get it working.

If you need something to reproduce see my [Test-Repo](https://github.com/do-me/Test) and install mkdocs-material 8.5 / 9 each.

**Reply by squidfunk** [reply]
Thanks for providing a little more details. If you could provide a minimal reproduction, or a PR that stats exactly what needs to be looked out for, we can discuss how we could improve the upgrade guide. Other than that it's impossible for me to anticipate _how_ authors override templates. I plan to add a tool to easily diff necessary template changes, which I think to be a better solution.

That said, until now, I have not seen any detailed error report on _what's_ actually wrong, so I'm unable to help. The minimum I need is a reproduction and a bug report.

**Reply by squidfunk** [reply]
Oh, and your repository [doesn't even mention the search plugin](https://github.com/do-me/Test/blob/944de693f2ca362bb58106f72fd6a52a5907f6a5/mkdocs.yml#L93-L99). [It must be enabled explicitly](https://squidfunk.github.io/mkdocs-material/setup/setting-up-site-search/#built-in-search-plugin) when defining `plugins`.

**Reply by do-me** [reply]
You're right, I referenced the wrong "Test"-repo. Sorry for that; I understand the effort and pain in hunting down bugs but only finding user errors. Will provide a real test repo soon.

**Reply by facelessuser** [reply]
IIRC, I had this happen as well. If you are overriding `header.html`, which I was and you seem to as well, I had to edit this:

```
    <!-- Button to open search modal -->
    {% if "material/search" in config.plugins %}
      <label class="md-header__button md-icon" for="__search">
        {% include ".icons/material/magnify.svg" %}
      </label>

      <!-- Search interface -->
      {% include "partials/search.html" %}
    {% endif %}
```

Specifically, it should be noted that prior to the recent release you had to look for `search` in the plugins, now it requires `material/search`. Hope that helps.

**Reply by facelessuser** [reply]
@squidfunk , this should be documented as I imagine others will run into this as well.

**Reply by do-me** [reply]
That's it @facelessuser! Thanks a lot. 
@squidfunk could you mark this as the answer for others? I guess the upgrade guide would be the right place to add a note about changing partials.

**Reply by squidfunk** [reply]
Ah, thanks, yeah, that's likely the thing everybody is tripping over. We've had this change for so long in Insiders, it didn't came to my mind that this might be the cause. I'll add something to the upgrade guide.

**Reply by squidfunk** [reply]
Added in 88cfdce58. The [upgrade guide](https://squidfunk.github.io/mkdocs-material/upgrade/#changes-to-html-files) now includes:

<img width="740" alt="Bildschirmfoto 2023-01-20 um 17 26 59" src="https://user-images.githubusercontent.com/932156/213751723-e043f3b4-cf77-4d16-9030-31ea4308ddce.png">

**Comment by SoumayaMauthoorMOJ** [comment]
Had the same problem! @squidfunk  Can I suggest you update https://github.com/squidfunk/mkdocs-material/blob/master/docs/setup/setting-up-site-search.md to:
```
plugins:
  - material/search
```

**Reply by squidfunk** [reply]
Thanks @SoumayaMauthoorMOJ. This is unnecessary – you can and should just configure it as `search`. It is only necessary if you have a second third-party plugin (not MkDocs own search plugin) installed that is also called `search`.

**Reply by SoumayaMauthoorMOJ** [reply]
Understood  @squidfunk 👍  Perhaps it should be included in https://squidfunk.github.io/mkdocs-material/customization? https://ministryofjustice.github.io/analytical-platform-data-engineering/ uses a theme that I have customised from mkdocs material (https://github.com/ministryofjustice/mkdocs-tech-docs-template) . I had to set the plugin to `material\search` in order for the search bar to appear, even though I don't think I have installed a second third-party plugin.
